enhancements/network/configurable-network-diagnostics-pod-placement.md

```diff
@@ -0,0 +1,289 @@
+}
+// NetworkDiagnostics defines network diagnostics configuration
+type NetworkDiagnostics struct {
+	// sourceDeploymentNodePlacement controls the scheduling of network diagnostics source deployment
+	//
+	// See NetworkDiagnosticsNodePlacement for more details about default values.
+	//
+	// +optional
+	SourceDeploymentNodePlacement NetworkDiagnosticsNodePlacement `json:"sourceDeploymentNodePlacement"`
+
+	// targetDaemonsetNodePlacement controls the scheduling of network diagnostics target daemonset
+	//
+	// See NetworkDiagnosticsNodePlacement for more details about default values.
+	//
+	// +optional
+	TargetDaemonsetNodePlacement NetworkDiagnosticsNodePlacement `json:"targetDaemonsetNodePlacement"`
+}
+
+// NetworkDiagnosticsNodePlacement defines node scheduling configuration network diagnostics components
+type NetworkDiagnosticsNodePlacement struct {
+	// nodeSelector is the node selector applied to a network diagnostics component
+	//
+	// By default this is set to `kubernetes.io/os: linux`
+	//
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector"`
+
+	// tolerations is a list of tolerations applied to a network diagnostics component
+	//
+	// For sourceDeploymentNodePlacement, this is set to an empty list by default.
+	//
+	// For targetDaemonsetNodePlacement, this is set to `- operator: "Exists"` by default.
+	//
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations"`
+}
+```
+The fields will initially be added as TechPreview and will move to Default once they meet the 
+graduation criteria.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+In Hypershift, the network diagnostics components do not run in the controlplane so there are no 
+special considerations.
+
+#### Standalone Clusters
+
+No special considerations.
+
+#### Single-node Deployments or MicroShift
+
+MicroShift doesn't run CNO and network diagnostics.  
+No special considerations for single-node deployments.
+
+### Implementation Details/Notes/Constraints
+
+This enhancement requires changes to the `network.operator.openshift.io` CRD and the CNO 
+operator. CNO will apply the network diagnostics settings on every reconciliation. This means 
+that they can be changed at any point during the cluster lifetime. 
+
+### Risks and Mitigations
+
+If there is no `network-check-source` pods running the network diagnostic feature is effectively 
+disabled. In OpenShift 4.15 and prior, this would lead to CNO setting the `Progressing` condition to
+`True` with a message stating that it is waiting for the deployment to succeed. This meant that CNO 
+would stay in this state forever, which is unexpected and misleading to the user.   
+This enhancement removes this behavior and to keep the discoverability, the CNO emits a warning 
+event instead.
+
+### Drawbacks
+
+Enabling the cluster administrator to affect the pod placement of the network diagnostic feature 
```

Comment:
is the network check application smart enough that it will only check the connectivity to nodes where the target pod is running, and not report alerts/errors if a target pod is not running  on a particular node?

Reply:
the connectivity check controller creates a `PodNetworkConnectivityCheck` objects for every target endpoint so this shouldn't be an issue